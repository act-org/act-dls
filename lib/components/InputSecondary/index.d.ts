/**
 * @prettier
 */
import * as React from 'react';
import { InputProps } from '@material-ui/core';
import { Size } from '~/types';
export declare type Props = InputProps & {
    size?: Size;
};
declare const InputSecondary: React.FC<Props>;
export default InputSecondary;
//# sourceMappingURL=index.d.ts.map