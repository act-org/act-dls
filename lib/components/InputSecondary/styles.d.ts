/**
 * @prettier
 */
declare const _default: (props?: any) => Record<"default" | "large" | "inputInput" | "inputInputPassword" | "inputRoot" | "multiline", string>;
export default _default;
//# sourceMappingURL=styles.d.ts.map