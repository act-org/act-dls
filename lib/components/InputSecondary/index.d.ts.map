{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/InputSecondary/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAG/B,OAAO,EAAS,UAAU,EAAE,MAAM,mBAAmB,CAAC;AAEtD,OAAO,EAAE,IAAI,EAAE,MAAM,SAAS,CAAC;AAI/B,oBAAY,KAAK,GAAG,UAAU,GAAG;IAAE,IAAI,CAAC,EAAE,IAAI,CAAA;CAAE,CAAC;AAEjD,QAAA,MAAM,cAAc,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAqCnC,CAAC;AAOF,eAAe,cAAc,CAAC"}