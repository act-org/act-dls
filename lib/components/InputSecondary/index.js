"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _clsx = _interopRequireDefault(require("clsx"));

var _Input = _interopRequireDefault(require("@material-ui/core/Input"));

var _mergeClasses = _interopRequireDefault(require("../../helpers/mergeClasses"));

var _styles = _interopRequireDefault(require("./styles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var InputSecondary = function InputSecondary(_ref) {
  var classesProp = _ref.classes,
      multiline = _ref.multiline,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'default' : _ref$size,
      type = _ref.type,
      value = _ref.value,
      otherProps = _objectWithoutProperties(_ref, ["classes", "multiline", "size", "type", "value"]);

  var classes = (0, _styles["default"])();
  return /*#__PURE__*/React.createElement(_Input["default"], _extends({
    classes: (0, _mergeClasses["default"])({
      input: (0, _clsx["default"])(classes.inputInput, type === 'password' && !!value && String(value).length > 0 && classes.inputInputPassword),
      root: (0, _clsx["default"])(classes.inputRoot, classes[size], multiline && classes.multiline)
    }, classesProp),
    disableUnderline: true,
    margin: "dense",
    multiline: multiline,
    type: type,
    value: value
  }, otherProps));
}; // eslint-disable-next-line immutable/no-mutation


InputSecondary.defaultProps = {
  size: 'default'
};
var _default = InputSecondary;
exports["default"] = _default;