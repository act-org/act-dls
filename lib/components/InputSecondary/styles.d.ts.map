{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/InputSecondary/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAQH,wBA6BG"}