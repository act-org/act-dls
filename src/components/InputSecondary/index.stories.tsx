/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';

import InputSecondary from '.';

export const Secondary = (): React.ReactElement<any> => {
  const [value, setValue] = React.useState('Input');

  return (
    <InputSecondary
      autoFocus={boolean('Auto Focus?', false)}
      disabled={boolean('Disabled?', false)}
      multiline={boolean('Multiple Lines?', false)}
      onChange={(e: any): void => {
        action('onChange()');

        setValue(e.target.value);
      }}
      placeholder={text('Placeholder', 'Placeholder')}
      size={select(
        'Size',
        {
          Default: 'default',
          Large: 'large',
        },
        'default',
      )}
      type={select(
        'Type',
        {
          Password: 'password',
          Text: 'text',
        },
        'text',
      )}
      value={value}
    />
  );
};

export default {
  component: InputSecondary,
  parameters: {
    info: {
      text: 'This is the secondary variant of the Input component.',
    },
  },
  title: 'Atoms/Input',
};
