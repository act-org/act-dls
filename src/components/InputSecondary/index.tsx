/**
 * @prettier
 */

import * as React from 'react';
import clsx from 'clsx';
import { get, omit } from 'lodash';
import { Input, InputProps } from '@material-ui/core';

import { Size } from '~/types';

import useStyles from './styles';

export type Props = InputProps & { size?: Size };

const InputSecondary: React.FC<Props> = ({
  classes: classesProp,
  multiline,
  size = 'default',
  type,
  value,
  ...otherProps
}: Props): React.ReactElement<any> => {
  const classes = useStyles();

  return (
    <Input
      classes={{
        input: clsx(
          classes.inputInput,
          type === 'password' &&
            value &&
            String(value).length > 0 &&
            classes.inputInputPassword,
          get(classesProp, 'input'),
        ),
        root: clsx(
          classes.inputRoot,
          classes[size],
          multiline && classes.multiline,
          get(classesProp, 'root'),
        ),
        ...(omit(classesProp, ['input', 'root']) || {}),
      }}
      disableUnderline
      margin="dense"
      multiline={multiline}
      type={type}
      value={value}
      {...otherProps}
    />
  );
};

// eslint-disable-next-line immutable/no-mutation
InputSecondary.defaultProps = {
  size: 'default',
};

export default InputSecondary;
