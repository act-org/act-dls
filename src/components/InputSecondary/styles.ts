/**
 * @prettier
 */

import { makeStyles } from '@material-ui/core/styles';

import COLORS from '~/constants/colors';
import DIMS from '~/constants/dims';
import TYPOGRAPHY from '~/constants/typography';

export default makeStyles({
  default: {
    height: DIMS.SECONDARY.HEIGHTS.INPUT,
    minHeight: DIMS.SECONDARY.HEIGHTS.INPUT,
  },
  inputInput: {
    '&.Mui-disabled': {
      cursor: 'not-allowed',
    },
    fontSize: 14,
    padding: `0 ${DIMS.LAYOUT_PADDING}px`,
  },
  inputInputPassword: {
    fontSize: 22,
    fontWeight: TYPOGRAPHY.WEIGHT_BOLD,
    letterSpacing: 4,
  },
  inputRoot: {
    border: `1px solid ${COLORS.BORDER_SECONDARY}`,
    borderRadius: DIMS.SECONDARY.BORDER_RADIUS * 2,
    color: COLORS.DARK_SECONDARY,
  },
  large: {
    height: DIMS.SECONDARY.HEIGHTS.INPUT + DIMS.SECONDARY.HEIGHT_INCREMENT,
    minHeight: DIMS.SECONDARY.HEIGHTS.INPUT + DIMS.SECONDARY.HEIGHT_INCREMENT,
  },
  multiline: {
    height: 'unset',
  },
});
