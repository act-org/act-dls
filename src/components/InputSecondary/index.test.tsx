/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import InputSecondary from '.';

describe('InputSecondary', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <InputSecondary onChange={noop} value="Value" />,
    );

    expect(container).toMatchSnapshot();
  });
});
