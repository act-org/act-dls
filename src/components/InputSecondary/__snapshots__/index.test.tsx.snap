// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSecondary matches the snapshot 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root makeStyles-inputRoot-4 makeStyles-default-1 MuiInputBase-marginDense MuiInput-marginDense"
  >
    <input
      class="MuiInputBase-input MuiInput-input makeStyles-inputInput-2 MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
      type="text"
      value="Value"
    />
  </div>
</div>
`;
